### DISCUSSION TITLE
Search Syntax

### DISCUSSION THREAD

**Discussion by drwb** [bpatrik/pigallery2#691] [Q&A] [answered]
Is there an overview of the syntax for constructing advanced searches available somewhere? It seems that some amazing elements have been implemented, but I'm unsure how to use them. For example it seems [from this code that](https://github.com/bpatrik/pigallery2/commit/d0153e58d9a45019bff7e9741a8ac53a3c3ab472) `!:` is for a negating query. I'm unsure how to do `or` operators, file sizes. Perhaps there is a way in the search to set the sort order of the results?

**Comment by bpatrik** [comment, marked as answer]
There is no documtation about it.

The best you can do is to open the search query builder.that should support all features:

![Screenshot_20230809-180652.png](https://github.com/bpatrik/pigallery2/assets/7512182/25c3eabf-0d9d-4def-8010-c65fa1c00223)

![Screenshot_20230809-180641.png](https://github.com/bpatrik/pigallery2/assets/7512182/b26e37ae-4ae5-4999-8952-781cc0b5baf1)

I would call out the 'some-of', the date pattern and 5-km-from search queries as they are less intuitive but cool features.

The rest is simple metadata filter and boolean logic operators. You can do negate and exact search (the string fully matches not only a part of the found field)

There is not possible to do server side sorting with search. You can do that on the client side. 


You can think of search like a filter.

**Reply by drwb** [reply]
Thank you for your considered response

**Comment by kagahd** [comment]
> There is no documtation about it.

Well, there is indeed a bit of documentation about `Advanced searching`, `Match types`, `Supported keywords`, `Bool expressions` and `Sort order` on your homepage, @bpatrik: http://bpatrik.github.io/pigallery2/

**Reply by bpatrik** [reply]
The idea/intention of the app is to be as self explanatory as possible. So I'm not really keen on doing any documentation.

**Reply by kagahd** [reply]
As I pointed out, you documented it already @bpatrik 😄
